frontend/js/tests/user/stats/UserArtistMap.test.tsx

```diff
@@ -1,247 +1,145 @@
-
-      await act(async () => {
-        instance.changeSelectedMetric("artist");
-      });
-      await waitForComponentToPaint(wrapper);
-      expect(wrapper.state()).toMatchObject({
-        data: userArtistMapProcessedDataArtist,
-        selectedMetric: "artist",
-      });
+  it("displays error message when API call fails", async () => {
+    renderWithProviders(
+      <UserArtistMap {...{ ...props, range: "month" }} />,
+      {},
+      {
+        wrapper: reactQueryWrapper,
+      }
+    );
+
+    // Wait for loading to finish
+    await waitFor(() => {
+      expect(screen.getByTestId("error-message")).toBeInTheDocument();
     });
 
-    it('sets state correctly for "listen"', async () => {
-      const wrapper = shallow<UserArtistMap>(<UserArtistMap {...props} />);
-      const instance = wrapper.instance();
-
-      instance.rawData = userArtistMapResponse as UserArtistMapResponse;
-
-      await act(async () => {
-        instance.changeSelectedMetric("listen");
-      });
-      await waitForComponentToPaint(wrapper);
-      expect(wrapper.state()).toMatchObject({
-        data: userArtistMapProcessedDataListen,
-        selectedMetric: "listen",
-      });
-    });
+    expect(screen.getByText("Failed to fetch data")).toBeInTheDocument();
+    expect(screen.queryByTestId("Choropleth")).not.toBeInTheDocument();
   });
 
-  describe("loadData", () => {
-    it("calls getData once", async () => {
-      const wrapper = shallow<UserArtistMap>(<UserArtistMap {...props} />);
-      const instance = wrapper.instance();
-
-      const spy = jest.fn();
-      instance.getData = spy;
-      instance.processData = jest.fn();
-      await act(async () => {
-        await instance.loadData();
-      });
-      await waitForComponentToPaint(wrapper);
-
-      expect(spy).toHaveBeenCalledTimes(1);
+  it("renders choropleth with processed data", async () => {
+    renderWithProviders(
+      <UserArtistMap {...props} />,
+      {},
+      {
+        wrapper: reactQueryWrapper,
+      }
+    );
+
+    // Wait for loading to finish
+    await waitFor(() => {
+      expect(screen.getByTestId("Choropleth")).toBeInTheDocument();
     });
 
-    it("set state correctly", async () => {
-      const wrapper = shallow<UserArtistMap>(<UserArtistMap {...props} />);
-      const instance = wrapper.instance();
-
-      instance.getData = jest
-        .fn()
-        .mockImplementationOnce(() => Promise.resolve(userArtistMapResponse));
-      await act(async () => {
-        await instance.loadData();
-      });
-      await waitForComponentToPaint(wrapper);
-
-      expect(instance.rawData).toMatchObject(userArtistMapResponse);
-
-      expect(wrapper.state()).toMatchObject({
-        data: userArtistMapProcessedDataArtist,
-        loading: false,
-      });
-    });
+    // Check if the choropleth is rendered correctly
+    const choropleth = screen.getByTestId("Choropleth");
+    expect(choropleth).toBeInTheDocument();
```

Comment:
Same comment as above, I think you can remove this line as it duplicates the assertion on line 136